src/components/TeamMemberTasks/ReviewButton.jsx

```diff
@@ -108,23 +108,17 @@ const ReviewButton = ({
         <ModalFooter>
           <Button
             onClick={(e) => {
-              reviewStatus == "Unsubmitted"
-              ? (updReviewStat("Submitted"),
-                sendReviewReq(e))
-              : updReviewStat("Reviewed");
+              const newStatus = reviewStatus === "Unsubmitted" ? "Submitted" : "Reviewed";
+              updReviewStat(newStatus);
+              sendReviewReq(e);
```

Comment:
This line of code recursively calls the `sendReviewReq` function.

Something seems to be wrong. If there's a need, will it not be better to invoke the function outside the component? A better practice will be to have the onClick function defined in a different block scope.

Reply:
Sorry, I dont understand. How is this recursive?

Reply:
Invoking the `sendReviewReq` function in the same component that is meant to invoke it.
By being recursive, **the function calls itself**.
It might work well now, but it may cause performance issues in the future.
It is much more better that there is seperation of concerns. 

The application logic for the `sendReviewReq` function that handles the event of the Button component in the Modal footer div; should be isolated from the `sendReviewReq` functional component.

Reply:
Pragmatically speaking:

```

const sendReviewBtn = event => {
event.preventDefault();
// other related logic goes here...
}

const sendReviewReq = () => {
<ModalFooter>
          <Button
            onClick={(e) => {
              const newStatus = reviewStatus === "Unsubmitted" ? "Submitted" : "Reviewed";
              updReviewStat(newStatus);
              sendReviewBtn(e); // Noticed that i changed the name of the function to match with what was declared above.
              }}
            color="primary"
            className="float-left"
            style={boxStyle}
          >
            {reviewStatus === "Unsubmitted" ? "Submit" : "Complete"}
          </Button>
// OTHER COMPONENTS AND JSX GOES HERE
           </ModalFooter>
}
```

Reply:
> Sorry, I dont understand. How is this recursive?

It's not, and it works fine.